### PR TITLE
Updated Upstream (Bukkit/CraftBukkit)

### PR DESCRIPTION
**Pull request by kennytv**
Upstream has released updates that appear to apply and compile correctly.
This update has not been tested by PaperMC and as with ANY update, please do your own testing

Bukkit Changes:
44cfe143 SPIGOT-6249: Add Missing Effect Constants

CraftBukkit Changes:
14928261 SPIGOT-6249: Add Missing Effect Constants
332335e1 SPIGOT-6731: "Nag author" message in CraftServer lists one author only
6cd975d0 SPIGOT-5732, SPIGOT-6387: Overhaul Hanging entities

### CODE REVIEW

File: patches/api/0274-Add-missing-effects.patch [hunk 1]
```diff
@@ -28,49 +28,111 @@ index d0c812c72967469122f7f276ced31cfb0e9482dc..0f598fa0560d14b44dcc65205364870c
 +    /**
 +     * The sound of phantom's bites
 +     */
++    @Deprecated
```
Comment (Proximyst) [inline patches/api/0274-Add-missing-effects.patch]: forRemoval

File: patches/api/0274-Add-missing-effects.patch [hunk 2]
```diff
@@ -28,49 +28,111 @@ index d0c812c72967469122f7f276ced31cfb0e9482dc..0f598fa0560d14b44dcc65205364870c
...
+@@ -1,13 +1,16 @@
+ package org.bukkit;
+ 
++import static org.bukkit.Effect.*; // Paper
+ import static org.hamcrest.CoreMatchers.*;
+ import static org.junit.Assert.*;
+ import org.junit.Test;
+ 
+ public class EffectTest {
++    private static final java.util.Set<Effect> DEPRECATED_PAPER_TYPES = java.util.EnumSet.of(PHANTOM_BITES, ZOMBIE_CONVERTS_TO_DROWNED, HUSK_CONVERTS_TO_ZOMBIE, GRINDSTONE_USED, BOOK_PAGE_TURNED, COMPOSTER_COMPOSTS, LAVA_CONVERTS_BLOCK, REDSTONE_TORCH_BURNS_OUT, ENDER_EYE_PLACED, ENDER_DRAGON_DESTROYS_BLOCK, WET_SPONGE_VAPORIZES_IN_NETHER); // Paper
```
Comment (Proximyst) [inline patches/api/0274-Add-missing-effects.patch]: Could probably either detect Deprecated or reflect if you disliked this

Comment (kennytv) [reply patches/api/0274-Add-missing-effects.patch]: Not pretty, but at least slightly safer!

File: patches/api/0274-Add-missing-effects.patch [hunk 3]
```diff
@@ -28,49 +28,131 @@ index d0c812c72967469122f7f276ced31cfb0e9482dc..0f598fa0560d14b44dcc65205364870c
 +    /**
 +     * The sound of phantom's bites
```
Comment (lynxplay) [inline patches/api/0274-Add-missing-effects.patch:29]: I guess for the sake of consistency, this one is missing the `@deprecated use @{link #PHANTOM_BITE}` comment.

File: patches/api/0274-Add-missing-effects.patch [hunk 4]
```diff
@@ -28,49 +28,131 @@ index d0c812c72967469122f7f276ced31cfb0e9482dc..0f598fa0560d14b44dcc65205364870c
...
+             BY_ID.put(effect.id, effect);
+         }
+     }
+ 
++    // Paper start
++    private static boolean isDeprecated(Effect effect) {
++        try {
++            return Effect.class.getDeclaredField(effect.name()).isAnnotationPresent(Deprecated.class);
++        } catch (NoSuchFieldException e) {
++            e.printStackTrace();
```
Comment (Proximyst) [inline patches/api/0274-Add-missing-effects.patch]: Make a private logger to log a proper error message

**Review by Proximyst — CHANGES_REQUESTED** [review]

**Pull request merged**